enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -0,0 +1,162 @@
+---
+title: api-removal-notifications
+authors:
+  - "@sanchezl"
+reviewers:
+  - "@deads2k"
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-01
+last-updated: 2021-02-01
+status: provisional
+---
+
+# API Removal Notifications
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Notify customers that an API that will be removed in the next release is in use.
+
+## Motivation
+
+Kubernetes is being more aggressive with the removal of APIs that have been deprecated or pre-release APIs that have
+reached a dead-end and will never reach GA. Additionally, the ability to re-enable a previously removed API has been
+removed.
+
+Customers should be notified that an API that will be removed in the next release is in use so that they can decide if
+there is a need to take action in order to ensure success when upgrading to the next release.
+
+### Goals
+
+1. Notify customers that an API that will be removed in the next release is in use.
+2. Notify EUS customers that an API that will be removed in the next EUS release is in use.
```

Comment:
If we get the three minor versions notification in point number one, then this just becomes backporting from data from 4.7 to 4.6 in the case of 4.6 and 4.10 being EUS versions.

Reply:
> If we get the three minor versions notification in point number one, then this just becomes backporting from data from 4.7 to 4.6 in the case of 4.6 and 4.10 being EUS versions.

I don't anticipate making that change, since it will produce noise that is unnecessary.  If you have a signal for determining a customer planning on using 4.6 as an EUS, then we could reconsider, but without such a signal looking three releases out will simply produce noise.

Reply:
> If you have a signal for determining a customer planning on using 4.6 as an EUS, then we could reconsider...

We have that signal today by looking for folks in the `eus-4.6` channel.  Although folks can always change channels. 
 Once `stable-4.6` goes EOL, anyone on later 4.6.z will have to be `eus-4.6` folks who will eventually either tear down their clusters or try to update out to at least 4.8, and giving those folks a bit of lead time on things which might block their transition into 4.8 seems a lot nicer than surprising them once they land in 4.7 (which they probably planned to pass through extremely quickly on their way from 4.6 to 4.8).